sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/NonAzureOpenAIKeyCredential.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Token credential class for accepting non-Azure OpenAI credential string.
+ */
+public class NonAzureOpenAIKeyCredential implements TokenCredential {
+    private final String keyCredential;
+
+    /**
+     * Create a non-Azure OpenAI credential token.
+     *
+     * @param keyCredential non-Azure OpenAI credential.
+     */
+    public NonAzureOpenAIKeyCredential(String keyCredential) {
+        this.keyCredential = keyCredential;
+    }
+
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext request) {
+        return Mono.fromCallable(() -> new AccessToken(keyCredential, OffsetDateTime.now().plusDays(180)));
```

Comment:
Why is the expiration set to 180 days? The key (`keyCredential`) doesn't change automatically after this period.

Reply:
I think Shawn is trying to mirror what we have .NET, but maybe there is a different way to achieve this: 

https://github.com/Azure/azure-sdk-for-net/blob/35863b12c2ac57911bf661ee8630852bb4727af2/sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs#L667

Reply:
Updated the credential prototype without extending from TokenCredential now. So there is no consideration for expiration time now. It is no longer an issue consideration. So closed this feedback.